machine/v1beta1/types_awsprovider.go

```diff
@@ -95,6 +95,17 @@ type AWSMachineProviderConfig struct {
 	// The field size should be greater than 0 and the field input must start with cr-***
 	// +optional
 	CapacityReservationID string `json:"capacityReservationId"`
+	// marketType specifies the type of market for the EC2 instance.
+	// Valid values are OnDemand, Spot, and CapacityBlock.
+	// Defaults to OnDemand.
+	// When set to OnDemand the instance runs as a standard OnDemand instance.
+	// When set to Spot the instance runs as a Spot instance. When SpotMarketOptions is provided, the marketType defaults to "Spot".
+	// When set to CapacityBlock the instance utilizes pre-purchased compute capacity (capacity blocks) with AWS Capacity Reservations.
+	// If this value is selected, CapacityReservationID must be specified to identify the target reservation.
+	// If marketType is not specified and spotMarketOptions is provided, the marketType defaults to "Spot".
```

Comment:
I think this is already covered by other parts of the godoc, can probably just remove it:
```suggestion
```